beefy-gadget/src/worker.rs

```diff
@@ -142,32 +140,18 @@ where
 {
 	/// Return `true`, if we should vote on block `number`
 	fn should_vote_on(&self, number: NumberFor<B>) -> bool {
-		use sp_runtime::{traits::Saturating, SaturatedConversion};
-
 		let best_beefy_block = if let Some(block) = self.best_beefy_block {
 			block
 		} else {
 			debug!(target: "beefy", "🥩 Missing best BEEFY block - won't vote for: {:?}", number);
 			return false;
 		};
 
-		let diff = self.best_grandpa_block.saturating_sub(best_beefy_block);
-		let diff = diff.saturated_into::<u32>();
-		let next_power_of_two = (diff / 2).next_power_of_two();
-		let next_block_to_vote_on = self.best_block_voted_on + self.min_block_delta.max(next_power_of_two).into();
-
-		trace!(
-			target: "beefy",
-			"should_vote_on: #{:?}, diff: {:?}, next_power_of_two: {:?}, next_block_to_vote_on: #{:?}",
-			number,
-			diff,
-			next_power_of_two,
-			next_block_to_vote_on,
-		);
+		let candidate = vote_candidate::<B>(number, self.best_grandpa_block, best_beefy_block, self.min_block_delta);
 
-		metric_set!(self, beefy_should_vote_on, next_block_to_vote_on);
+		metric_set!(self, beefy_should_vote_on, candidate);
 
-		number == next_block_to_vote_on
+		number == candidate
```

Comment:
This line will prevent to vote on a candidate which is ahead of `number`.

Reply:
Yeah, renaming might be good. To me it's basically: `next_block_to_vote_on`.